test/parallel/test-http2-server-push-priority.js

```diff
@@ -0,0 +1,46 @@
+'use strict';
+
+const common = require('../common');
+if (!common.hasCrypto)
+common.skip('missing crypto');
+const assert = require('assert');
+const http2 = require('http2');
+
+const expectedWeight = 256;
+
+const server = http2.createServer();
+server.on('stream', common.mustCall((stream, headers, flags) => {
+  const options = { weight: expectedWeight };
+  stream.pushStream({}, options, common.mustCall((stream, headers, flags) => {
+    // The priority of push streams is changed silently by nghttp2,
+    // without emitting a PRIORITY frame. Flags are ignored.
+    assert.strictEqual(stream.state.weight, expectedWeight);
+    // assert.ok(flags & http2.constants.NGHTTP2_FLAG_PRIORITY);
+  }));
+  stream.respond({
+    'content-type': 'text/html',
+    ':status': 200
+  });
+  stream.end('test');
+}));
+
+server.listen(0, common.mustCall(() => {
+  const port = server.address().port;
+  const client = http2.connect(`http://localhost:${port}`);
+
+  const headers = { ':path': '/' };
+  const req = client.request(headers);
+
+  client.on('stream', common.mustCall((stream, headers, flags) => {
+    // Since the push priority is set silently, the client is not informed.
```

Comment:
there is a priority event that *should* be emitted, I believe.

Reply:
The issue here is that on the wire I wasn't seeing a priority flag on the headers frame (push request), nor a separate priority frame. This is different from client-initiated streams.

Reply:
From the nghttp2 docs:
> [nghttp2_submit_push_promise](https://www.nghttp2.org/documentation/nghttp2_submit_push_promise.html#c.nghttp2_submit_push_promise) [...]
> The flags is currently ignored.